signin.js

```diff
@@ -0,0 +1,74 @@
+const myInputEmail = document.querySelector('.input-email');
+const myInputPassword = document.querySelector('.input-password');
+const errorMessageId = document.getElementById('error-message-id');
+const errorMessagePw = document.getElementById('error-message-pw');
+const login = document.querySelector('.cta');
+const answerEmail = 'test@codeit.com';
+const answerPassword = 'codeit101';
+const toggleButtonPw = document.querySelector('.eye-button');
+const eyeOn = document.getElementById('eye-on');
+const eyeOff = document.getElementById('eye-off');
+
```

Comment:
js 파일을 만들면서, 생각보다 너무 많은 변수 선언에 당황스러웠습니다. 개수를 줄이면서 직관적으로 js 파일을 정리할 수 있는 방법이 있을 지 궁금합니다.

Reply:
제 개인적으로는 변수 선언이 많은 것은 중요하지 않다고 생각해요! 단, 우리가 지금 보면 모든 변수가 파일 최상단에 선언되어 모든 함수내에서 접근이 가능한 상황인데 특정 함수내에서만 사용한다면 해당 변수는 특정함수내로 가져가는게 좋아보여요! 

예를 들어 현재 아래와 같이 `answerEmail` , `answerPassword`의 경우에는 login의 onClick 이벤트안에 callback 함수에서만 사용되게 되는데 최상위에 선언했기 떄문에 다른 함수내에서도 접근이 가능하게 됩니다. 해당 부분을 필요한 함수내로 옮기게 된다면 해당 callback 함수의 필요한 인자로써 좀 더 응집도 높은 코드가 될 것 같아요! 

현재 코드상에서는 아래와 같은 형식을 갖추게 될 것 같아요. 

```suggestion
login.addEventListener('click', function() { 
 const answerEmail = 'test@codeit.com';
 const answerPassword = 'codeit101';
  if (myInputEmail.value === answerEmail && myInputPassword.value == answerPassword) {
    window.location.href = '/folder';
  } else {
    errorMessageId.textContent = '이메일을 확인해 주세요.';
    errorMessageId.style.display = 'block';
    myInputEmail.classList.add('error');
    errorMessagePw.textContent = '비밀번호를 확인해 주세요.';
    errorMessagePw.style.display = 'block';
    myInputPassword.classList.add('error');
  }
});

```